src/tribler-common/tribler_common/sentry_reporter/sentry_mixin.py

```diff
@@ -6,7 +8,13 @@ class AddBreadcrumbOnShowMixin:
     and send breadcrumbs on a show event.
     """
 
+    def __init__(self):
+        self.sentry_reporter: Optional[SentryReporter] = None
+
+    def set_sentry_reporter(self, sentry_reporter: SentryReporter):
```

Comment:
Question: is it really necessary to set the reporter every time? Why not use the default global instance  `gui_ sentry_reporter` directly?

Reply:
Yes, it was necessary because `SentryMixin` was placed in the `Common` package and `sentry_reporter` was placed in the `GUI` package.

Now `SentryMixin` has been moved to the `GUI` package. This should address your question.